modules/riemann/src/main/java/com/spotify/ffwd/riemann/RiemannOutputPlugin.java

```diff
@@ -36,25 +35,24 @@
 import com.spotify.ffwd.protocol.RetryPolicy;
 import java.util.Optional;
 
-public class RiemannOutputPlugin implements OutputPlugin {
+public class RiemannOutputPlugin extends OutputPlugin {
     private static final ProtocolType DEFAULT_PROTOCOL = ProtocolType.TCP;
     private static final int DEFAULT_PORT = 5555;
     private static final long DEFAULT_FLUSH_INTERVAL = 0;
-    // TimeUnit.MILLISECONDS.convert(10, TimeUnit.SECONDS);
 
-    private final Filter filter;
-    private final Long flushInterval;
     private final Protocol protocol;
     private final Class<? extends ProtocolClient> protocolClient;
     private final RetryPolicy retry;
 
     @JsonCreator
     public RiemannOutputPlugin(
-        @JsonProperty("filter") Filter filter, @JsonProperty("flushInterval") Long flushInterval,
-        @JsonProperty("protocol") ProtocolFactory protocol, @JsonProperty("retry") RetryPolicy retry
+        @JsonProperty("flushInterval") Optional<Long> flushInterval,
+        @JsonProperty("protocol") ProtocolFactory protocol,
+        @JsonProperty("retry") RetryPolicy retry, @JsonProperty("filter") Optional<Filter> filter
+
     ) {
-        this.filter = Optional.ofNullable(filter).orElseGet(TrueFilter::new);
-        this.flushInterval = Optional.ofNullable(flushInterval).orElse(DEFAULT_FLUSH_INTERVAL);
+        super("", filter,
+            flushInterval.isPresent() ? flushInterval : Optional.of(DEFAULT_FLUSH_INTERVAL));
```

Comment:
This seems wrong. `DEFAULT_FLUSH_INTERVAL == 0` has the same effect as being absent.

Somewhat unrelated to what you did, but since you are mucking around, you can fix it.